Automattic/wp-calypso
# Jetpack Backup: On the Backup page, show when there is a backup in progress

#### Changes proposed in this Pull Request

* Add `BackupInProgress` and `BackupJustCompleted` to the range of Daily Backup Status variants.
* Poll for the status of recent/in-progress backups when loading `DailyBackupStatus`.
* Move some backup status-related hooks around for easier re-use (e.g., `useDateWithOffset`, `useGetDisplayDate`).
* Create some shared selectors in the `rewind` hierarchy for monitoring realtime backup capability and in-progress backups.
* Add some more documentation around the `keepLocalTime` argument to `applySiteOffset`, to explain how it's used.

Fixes `1164141197617539-as-1199143164068597`.

#### Testing instructions

**Important:** Permutations you'll need to test for this PR:
* Calypso Blue and Calypso Green
* `jetpack/backup-simplified-screens-i4` feature flag on and off
* Desktop and mobile screen layouts
* Sites with and without existing backups
* As a user with an English locale and with a non-English locale.

1. Select the site with which you'll be testing and go to the Backup page.
2. Manually queue a backup for the site you selected, or wait for one to start.
3. Refresh the page.
4. Verify that you see a progress bar for the currently-running backup, and that the progress bar updates as the backup operation completes (see screenshots).
5. Navigate to a previous date while the backup is in progress. Verify that you see the correct status for the previous date, not the in-progress state.
6. Navigate back to the current date while the backup is still in progress. Verify you can still see the progress bar and that it still updates correctly.
7. Verify that when the backup completes, the page state changes to say that a backup was just completed and should be accessible in a few minutes. The timestamp of the backup should match the one from the "in progress" state.
8. Verify that if a previous backup existed before this one completed, you see a message that says, "Your last backup before today was ..." This text should link to the last successful backup before the current date.
9. Verify that if no previous backup existed beforehand, no text about "Your last backup before today" is shown.

#### Reference screenshots

**NOTES:**
* In some of the below images, there is no space before the sentence, "Your last backup before today was ..." This has been fixed.
* The date in some mobile layout shots below appears to be widowed. This has also been fixed.

##### Default/active variant

<img height="320" alt="Screen Shot 2020-12-21 at 08 41 30" src="https://user-images.githubusercontent.com/670067/102789536-e787dd00-4369-11eb-9b26-c51c46a22e6e.png"> <img height="320" alt="Screen Shot 2020-12-21 at 08 41 20" src="https://user-images.githubusercontent.com/670067/102789535-e787dd00-4369-11eb-9002-21122d8c1b5a.png">

<img height="320" alt="Screen Shot 2020-12-21 at 08 43 01" src="https://user-images.githubusercontent.com/670067/102789539-e8207380-4369-11eb-928d-90495a75b2ae.png"> <img height="320" alt="Screen Shot 2020-12-21 at 08 42 48" src="https://user-images.githubusercontent.com/670067/102789537-e787dd00-4369-11eb-9a19-a5d77b8974d5.png">

<img height="320" alt="Screen Shot 2020-12-21 at 08 44 12" src="https://user-images.githubusercontent.com/670067/102789541-e8207380-4369-11eb-850e-6ce334b0c880.png"> <img height="320" alt="Screen Shot 2020-12-21 at 08 44 03" src="https://user-images.githubusercontent.com/670067/102789540-e8207380-4369-11eb-8b4e-35b9294f6d7e.png">

<img height="320" alt="Screen Shot 2020-12-21 at 08 44 40" src="https://user-images.githubusercontent.com/670067/102789544-e8b90a00-4369-11eb-9559-4578e0f7e516.png"> <img height="320" alt="Screen Shot 2020-12-21 at 08 44 27" src="https://user-images.githubusercontent.com/670067/102789543-e8b90a00-4369-11eb-9e09-538557fafce8.png">

##### `jetpack/backup-simplified-screens-i4` variant (currently inactive)

<img height="320" alt="Screen Shot 2020-12-21 at 08 46 04" src="https://user-images.githubusercontent.com/670067/102790061-92000000-436a-11eb-898b-39419d7b9f8e.png"> <img height="320" alt="Screen Shot 2020-12-21 at 08 45 49" src="https://user-images.githubusercontent.com/670067/102790060-91676980-436a-11eb-948c-b40c255cfa47.png">

<img height="320" alt="Screen Shot 2020-12-21 at 08 48 13" src="https://user-images.githubusercontent.com/670067/102790064-92989680-436a-11eb-929d-a0e6b3facafb.png"> <img height="320" alt="Screen Shot 2020-12-21 at 08 47 57" src="https://user-images.githubusercontent.com/670067/102790062-92000000-436a-11eb-8b9d-136356be4fb7.png">

<img height="320" alt="Screen Shot 2020-12-21 at 08 51 29" src="https://user-images.githubusercontent.com/670067/102790067-92989680-436a-11eb-93fc-250a2a0cb1c7.png"> <img height="320" alt="Screen Shot 2020-12-21 at 08 51 18" src="https://user-images.githubusercontent.com/670067/102790065-92989680-436a-11eb-86ef-1726aab9b6b6.png">

<img height="320" alt="Screen Shot 2020-12-21 at 08 52 01" src="https://user-images.githubusercontent.com/670067/102790070-93312d00-436a-11eb-84d6-eb71f50242ea.png"> <img height="320" alt="Screen Shot 2020-12-21 at 08 51 48" src="https://user-images.githubusercontent.com/670067/102790069-92989680-436a-11eb-8b63-9ca15a7fb9b4.png">
